api/authenticator.py

```diff
@@ -733,6 +768,36 @@ def update_patron_metadata(self, patron):
         remote_patron_info = self.remote_patron_lookup(patron)
         if isinstance(remote_patron_info, PatronData):
             remote_patron_info.apply(patron)
+        if self.external_type_regular_expression:
+            self.update_patron_external_type(patron)
+
+    def update_patron_external_type(self, patron):
+        """Make sure the patron's external type reflects
+        what external_type_regular_expression says.
+        """
+        if not self.external_type_regular_expression:
+            # External type is not determined by a regular expression.
+            return
+        if not patron.authorization_identifier:
+            # Patron has no authorization identifier. Leave their
+            # external_type alone.
+            return
+
+        match = self.external_type_regular_expression.search(
+            patron.authorization_identifier
+        )
+        if not match:
+            # Patron's authorization identifier doesn't match the
+            # regular expression at all. Leave their external_type
+            # alone.
+            return
+        groups = match.groups()
+        if not groups:
+            # The regular expression matched but didn't contain any groups.
+            # This is a configuration error; do nothing.
+            return
+            
+        patron.external_type = groups[0]
```

Comment:
Is it true that there doesn't need to be a migration to create external_type for patrons that were using the regexp before because this code will create it the next time it's needed?